File: tests/cases/check/forbidden-package-references/lua/inputs/src/catalog/book.lua
local Order = require("orders.order")

-- catalog -> orders. In Lua a metatable base is the reference the model can
-- see: fields carry no type annotation, so an attribute never resolves.
local Book = setmetatable({}, { __index = Order })
Book.__index = Book

function Book.new(title)
  local self = setmetatable({}, Book)
  self.title = title
  return self
end

function Book:describe() return self.title end

return Book

File: tests/cases/check/forbidden-package-references/csharp/inputs/src/Catalog/Book.cs
using System.Collections.Generic;
using Orders;

namespace Catalog;

// Catalog -> Orders. `List<Order>` is unwrapped to `Order`.
public class Book
{
    public string Title { get; }
    public List<Order> Orders { get; }

    public Book(string title) { Title = title; Orders = new List<Order>(); }
}
